recipes/_install-windows.rb

```diff
@@ -19,6 +19,36 @@
 
 dd_agent_version = Chef::Datadog.agent_version(node)
 
+# TODO: make this dynamic
+BIN_PATH = 'C:/Program Files/Datadog/Datadog Agent/bin/agent'.freeze
+
+def fetch_current_version
+  return nil unless File.exist?(BIN_PATH)
+
+  agent_status = `"#{BIN_PATH}" status`
+  match_data = agent_status.match(/^Agent \(v(.*)\)/)
+
+  Gem::Version.new(match_data[1]) if match_data
+end
+
+def requested_agent_version
+  version = Chef::Datadog.agent_version(node)
+  Gem::Version.new(version) if version
+end
+
+def chef_version_uninstall_correctly?
```

Comment:
Can you add a comment explaining why we need to check this? That is: Chef < 14 not checking the right registry entry.